### PR TITLE
Fix neutron volume function.

### PR DESCRIPTION
**Pull request by ldgaetano**
# Description

I am trying to fix the neutron volume function and calculation within the ErgoScript contract + the offchain code. While doing this, I also found bug with how the proton volume was being calculated for the beta decay plus tests, so I have also corrected this. Current issue is that since the neutron volume calculation has changed, the expected value used in the tests is incorrect, these need to be updated so the tests can pass properly.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

I have modified the existing tests to account for the changes applied to the new calculation of the neutron volume.

#### To be composed in the future

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: modules/gluonw-base/src/main/resources/ErgoContracts/GluonW/Ergs/BoxGuardScripts/GluonWBoxGuardScript.es [hunk 1]
```diff
@@ -500,7 +501,7 @@
             val currentBlockNumber: Long = CONTEXT.HEIGHT
 
             // Check Protons reduction in OutBox
-            val worthOfMInErgs: BigInt = getProtonsPrice(M)
```
[the remark this answers is not in the document]
Comment (ldgaetano) [reply modules/gluonw-base/src/main/resources/ErgoContracts/GluonW/Ergs/BoxGuardScripts/GluonWBoxGuardScript.es:503]: yeah thought i had changed all occurences

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala [hunk 2]
```diff
@@ -103,15 +105,27 @@ case class GluonWConstants(precision: Long = GluonWBoxConstants.PRECISION)
...
   def neutronsToNanoErg(
+    neutronsInCirculation: Long,
     neutronsAmount: Long,
+    fissionedErg: Long,
     goldPriceGramsNanoErg: Long
-  ): Long =
-    (BigInt(neutronsAmount) * BigInt(goldPriceGramsNanoErg) / GluonWBoxConstants.PRECISION).toLong
+  ): Long = {
+    val fusRation: BigInt =
+      fusionRatio(
```
[the remark this answers is not in the document]
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:114]: Ok. Equation in paper did not change as far as I know, it just was not implemented in code the same as the paper.

File: modules/gluonw-base/src/main/resources/ErgoContracts/GluonW/Ergs/BoxGuardScripts/GluonWBoxGuardScript.es (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldgaetano) [reply modules/gluonw-base/src/main/resources/ErgoContracts/GluonW/Ergs/BoxGuardScripts/GluonWBoxGuardScript.es:503]: Done

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:114]: susRatio

[the remark this answers is not in the document]
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:114]: Done

File: modules/gluonw-base/src/main/scala/gluonw/boxes/GluonWBox.scala [hunk 3]
```diff
@@ -184,7 +192,7 @@ case class OracleBox(
         ("name", Json.fromString(GluonWAsset.NEUTRON.toString)),
         ("tokenId", Json.fromString(tokens.tail.head.getId.toString)),
         ("epochId", Json.fromLong(getEpochId.toLong)),
-        ("priceInNanoErgPerKg", Json.fromLong(getPrice))
+        ("priceInNanoErgPerGrams", Json.fromLong(getPricePerGrams))
```
Comment (ceilican) [inline modules/gluonw-base/src/main/scala/gluonw/boxes/GluonWBox.scala:195]: "PerGrams" => "PerGram"

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala [hunk 4]
```diff
@@ -103,15 +105,33 @@ case class GluonWConstants(precision: Long = GluonWBoxConstants.PRECISION)
...
 
   def neutronsToNanoErg(
+    neutronsInCirculation: Long,
     neutronsAmount: Long,
+    fissionedErg: Long,
     goldPriceGramsNanoErg: Long
-  ): Long =
-    (BigInt(neutronsAmount) * BigInt(goldPriceGramsNanoErg) / GluonWBoxConstants.PRECISION).toLong
+  ): Long = {
+    val fusRation: BigInt =
```
Comment (ceilican) [inline modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:113]: "Ration" => "Ratio"

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala [hunk 5]
```diff
@@ -103,15 +105,33 @@ case class GluonWConstants(precision: Long = GluonWBoxConstants.PRECISION)
...
+      fusionRatio(
+        neutronsInCirculation,
+        goldPriceGramsNanoErg,
+        fissionedErg
+      )
+
+    // Equations for determining the neutron price, Pn, and neutron volume, Vn.
```
Comment (ceilican) [inline modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:120]: These commnets are very nice. I suggest adding the to the contract as well.

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala [hunk 6]
```diff
@@ -122,11 +142,15 @@ case class GluonWConstants(precision: Long = GluonWBoxConstants.PRECISION)
         fissionedErg
       )
 
+    // Equations for determining the proton price, Pp, and proton volume, Vp.
```
Comment (ceilican) [inline modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala:145]: These comments too.

File: modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala [hunk 7]
```diff
@@ -166,10 +171,11 @@ class GluonWFeesSpec extends GluonWBase {
...
 
       assert(feeOutboxes.size == 3)
-      checkGluonWFeesBox(feeOutboxes.head, devFee, devAddress)
-      checkGluonWFeesBox(feeOutboxes.tail.head, uiFee, uiAddress)
+
+      checkGluonWFeesBox(feeOutboxes.tail.head, devFee, devAddress)
+      checkGluonWFeesBox(feeOutboxes.tail.tail.head, uiFee, uiAddress)
       checkGluonWFeesBox(
-        feeOutboxes.tail.tail.head,
+        feeOutboxes.head,
```
Comment (ceilican) [inline modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala:178]: This change is surprising. Could you tell me why this change had to introduced?

File: modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala [hunk 8]
```diff
@@ -182,20 +188,26 @@ class GluonWFeesSpec extends GluonWBase {
...
         )
       ) * gluonWFeesCalculator.uiFeesNumerator / gluonWFeesCalculator.denominator).toLong
       val oracleFee: Long = (BigInt(
         gluonWConstants.neutronsToNanoErg(
-          neutronsAmount,
-          oracleBox.getPricePerGrams
+          neutronsInCirculation = gluonWBox.neutronsCirculatingSupply,
+          neutronsAmount = neutronsAmount,
+          fissionedErg = gluonWBox.ergFissioned,
+          goldPriceGramsNanoErg = oracleBox.getPricePerGrams
```
Comment (ceilican) [inline modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala:210]: "PerGrams" => "PerGram"

File: modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala [hunk 9]
```diff
@@ -211,10 +223,10 @@ class GluonWFeesSpec extends GluonWBase {
         gluonWFeesCalculator.getFeesOutBox(betaDecayMinusFees)
 
       assert(feeOutboxes.size == 3)
-      checkGluonWFeesBox(feeOutboxes.head, devFee, devAddress)
-      checkGluonWFeesBox(feeOutboxes.tail.head, uiFee, uiAddress)
+      checkGluonWFeesBox(feeOutboxes.tail.head, devFee, devAddress)
```
Comment (ceilican) [inline modules/gluonw-base/src/test/scala/gluonw/common/GluonWFeesSpec.scala:226]: This is surprising too.

**Review by ceilican — CHANGES_REQUESTED** [review]

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala (hunk 5 above)
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala]: Ok, I added them.

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala (hunk 4 above)
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala]: Done.

File: modules/gluonw-base/src/main/scala/gluonw/boxes/GluonWBox.scala (hunk 3 above)
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/boxes/GluonWBox.scala]: Done

File: modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala (hunk 6 above)
Comment (ldgaetano) [reply modules/gluonw-base/src/main/scala/gluonw/common/GluonWAlgorithm.scala]: Done.

**Review by ceilican — APPROVED** [review]
